core/src/memory.rs

```diff
@@ -660,9 +662,7 @@ impl AllocationsContext {
     ///
     /// Currently running program should own this page.
     pub fn free(&mut self, page: WasmPage) -> Result<(), Error> {
-        if page > self.max_pages {
-            Err(Error::OutOfBounds)
-        } else if page < self.static_pages || !self.allocations.remove(&page) {
+        if page < self.static_pages || page > self.max_pages || !self.allocations.remove(&page) {
```

Comment:
```suggestion
        if page < self.static_pages || page >= self.max_pages || !self.allocations.remove(&page) {
```

Reply:
pages start with 0 ind, so max_pages, that means amount is already out of bounds

Reply:
Yes but OutOfBounds means that some body want to read or write access this pages - here we wanna just free it, so this is InvalidFree in first place

Reply:
Done